fix: missing validation for min number

theskupi opened the pull request:
There was already a `min` prop in DateInput component, but it wasn't doing anything. I changed it to number and added validation for it.

samuelg0rd0n reviewed (changes requested):
I like this general approach 😊 Thank you but since we're adding `min` prop, let's **please add also `max` prop** in `NumberInput` component and then set `max` to 31 in day field of `DateInput`.

I was also thinking if we shouldn't somehow include the actual number in the error message (in the form of a placeholder) because something like "Number is below minimum" is kind of ambiguous. Whereas "Number must be 1 or higher" is much more clear IMO. So the error message would be for example "Number must be __MIN__ or higher" and in the code you'll replace the placeholder. And then in the app, the user can either choose to completely ignore this and just use their own phrase or they can leverage this placeholder in the translations. In this case, it would be nice to put a comment for these message props. What do you think? Overkill? Well, maybe, just a thought 😃 (We can discuss on Slack 😃 )

theskupi commented:
@samuelg0rd0n Thanks for the points, good ideas! 🙇 I added max prop, amended the error messages and refactored condition in useEffect hook, because it started to look messy.
